### ISSUE TITLE
Fix plot_his() problem of outputing array.

### ISSUE THREAD

**Issue by tiffanylin43** [SylviaCHS/BME547_Final_Project#48]

**Comment by tiffanylin43** [comment]
@kkl18 Hi, Kim! I've edited the plot_his() function. You should be able use .tolist() to turn both his and bins into list that can be saved.

**Comment by KimberlyLennox** [comment]
@tiffanylin43 Hi Tiffany. Unfortunately, this still isn't fixed. Even after using .tolist() my code is trying to save this as an array. Please try saving the data into a json file on a Mongo database. I'll keep trying to fix this but I'm at a loss at this point.

**Comment by tiffanylin43** [comment]
@kkl18 Hi, Kim! Sylvia said that she've fixed this last night. Did you test the codes in her new branch? I think it's called "fix_get_image" with her netid. Let me know if this works.

**Comment by KimberlyLennox** [comment]
@tiffanylin43 @SylviaCHS I merged the fix_get_his branch into mine and checked the code myself to make sure that it merged correctly. Using .tolist() still does not work, nor does list(). In the end, I had to export every single bin and histogram separately as its own list and it still isn't working, but now the histograms are saving, just not the bins.

Has anyone else been able to successfully implement the server code complete with saving the histogram to the database?

**Comment by tiffanylin43** [comment]
@kkl18 Hi, Kim! I checked the pull request, it seems that you didn't merge Sylvia's latest codes which should have solved the problem. Her branch is "Hc239/client fix get image". Could you please check if those codes work?

**Comment by SylviaCHS** [comment]
@kkl18  Hi Kim, I solved the problem by changing Tiffany's image processing code. The latest version is on the branch hc239/client_fix_get_image. You can merge the branch and see if it's working now.

Thanks,
Sylvia

**Comment by KimberlyLennox** [comment]
@SylviaCHS 

Did this work in your version of the client code? I ran everything as-is and it's still broken

To clarify: I've merged fix_get_image into my own branch and then run the client code. It's identical to what's on GitHub. The same errors are coming back.

**Comment by SylviaCHS** [comment]
@kkl18 Yes, it is working. I print the type of hist and bins, they are 'list' now. I am not calling the them to plot histograms in GUI, but they shouldn't give an error.
![image](https://user-images.githubusercontent.com/46573933/56856436-4d6b5980-6928-11e9-89d5-201b487740d7.png)

What error are you getting? Have you change mongo.py? I am testing using my own database now.

**Comment by KimberlyLennox** [comment]
@SylviaCHS 
Hi Sylvia,

I had to make one more change to the image processing code to get the histogram data to save, but I still don't know if it will plot correctly.

The problem is that the function is making a list of numpy arrays and that seems to throw off the .tolist() function.

The only thing that I can think of is that it's an OS or a virtual environment issue. 

I saw the change in the database code. The original database has sped up considerably, but it will probably help if we aren't all querying it at the same time.
